src/App.tsx

```diff
@@ -1,98 +1,62 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Autocomplate } from './component/Autocomplete';
+import { debounce } from './utils/debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const currentPerson = peopleFromServer.find(
+    person => person.name === selectedPerson,
+  );
+
+  const handleQueryClick = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    setQuery(event.currentTarget.textContent || '');
+    setSelectedPerson(event.currentTarget.textContent || '');
+    setIsInputFocused(true);
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setIsInputFocused(true);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery]);
```

Comment:
move all logic except selectedPerson state in Autocomplete component.
Also pay attention on this point from description:
`the delay should be customizable via props (default value is 300ms);`
it means that delay should come to Autocomplet component as not required prop with default value 300ms.
For better testing pass delay 1000ms.